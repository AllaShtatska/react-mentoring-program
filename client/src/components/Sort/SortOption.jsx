import inlineStyles from "../../static/inline.css";
import sortButtonStyles from "../../static/sort-button.css";

import React from "react";
import SORT_TYPE from "./sortType";

class SortOption extends React.Component {
  get className() {
    const isSortOptionSelected =
      this.props.currentSortType === this.props.sortType;

    return isSortOptionSelected
      ? sortButtonStyles["sort-button-selected"]
      : sortButtonStyles["sort-button-deselected"];
  }

  getSortOptionTitle() {
    switch (this.props.sortType) {
      case SORT_TYPE.rating:
        return "rating";
      case SORT_TYPE.releaseDate:
        return "release date";
    }
  }

  render() {
    return (
      <div className={inlineStyles.inline}>
        <button
          className={this.className}
          onClick={() => this.props.onSwitchSortType(this.props.sortType)}
        >
          {this.getSortOptionTitle()}
        </button>
      </div>
    );
  }
}

export default SortOption;
