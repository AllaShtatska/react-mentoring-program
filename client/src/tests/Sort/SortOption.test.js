import React from "react";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";
import SortOption from "../../components/Sort/SortOption";
import SORT_TYPE from "../../components/Sort/sortType";

it("renders correctly if option rating is selected", () => {
  const component = renderer.create(<SortOption currentSortType={SORT_TYPE.rating} sortType={SORT_TYPE.rating} onSwitchSortType={function () {}}/>);

  const jsonComponent = component.toJSON();
  expect(jsonComponent).toMatchSnapshot();
});

it("renders correctly if option rating is not selected", () => {
  const component = renderer.create(<SortOption currentSortType={SORT_TYPE.releaseDate} sortType={SORT_TYPE.rating} onSwitchSortType={function () {}}/>);

  const jsonComponent = component.toJSON();
  expect(jsonComponent).toMatchSnapshot();
});

it("renders correctly if option release date is selected", () => {
  const component = renderer.create(<SortOption currentSortType={SORT_TYPE.releaseDate} sortType={SORT_TYPE.releaseDate} onSwitchSortType={function () {}}/>);

  const jsonComponent = component.toJSON();
  expect(jsonComponent).toMatchSnapshot();
});

it("renders correctly if option release date is not selected", () => {
  const component = renderer.create(<SortOption currentSortType={SORT_TYPE.rating} sortType={SORT_TYPE.releaseDate} onSwitchSortType={function () {}}/>);

  const jsonComponent = component.toJSON();
  expect(jsonComponent).toMatchSnapshot();
});

it("should trigger switch sort type on click", () => {
  const switchSortType = jest.fn();
  const component = shallow(<SortOption currentSortType={SORT_TYPE.rating} sortType={SORT_TYPE.releaseDate} onSwitchSortType={switchSortType}/>);

  component.find("button").simulate("click");

  expect(switchSortType).toHaveBeenCalled();
  expect(switchSortType).toBeCalledWith(SORT_TYPE.releaseDate);
});
