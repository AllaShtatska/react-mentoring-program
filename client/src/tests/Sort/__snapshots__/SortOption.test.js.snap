// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly if option rating is not selected 1`] = `
<div
  className="inline"
>
  <button
    className="sort-button-deselected"
    onClick={[Function]}
  >
    rating
  </button>
</div>
`;

exports[`renders correctly if option rating is selected 1`] = `
<div
  className="inline"
>
  <button
    className="sort-button-selected"
    onClick={[Function]}
  >
    rating
  </button>
</div>
`;

exports[`renders correctly if option release date is not selected 1`] = `
<div
  className="inline"
>
  <button
    className="sort-button-deselected"
    onClick={[Function]}
  >
    release date
  </button>
</div>
`;

exports[`renders correctly if option release date is selected 1`] = `
<div
  className="inline"
>
  <button
    className="sort-button-selected"
    onClick={[Function]}
  >
    release date
  </button>
</div>
`;
